### DISCUSSION TITLE
Between operator

### DISCUSSION THREAD

**Discussion by giorgi02** [dotnet/csharplang#6102] [Language Ideas] [answered]
It would be nice to have a comparison operator "Between" in C# in the same way as it is in Python
For example

Instead of:

```
int x = 5;
if (1 <= x && x <= 10)
    // Code
```

Write:

```
int x = 5;
if (1 <= x <= 10)
    // Code
```

**Comment by hez2010** [comment]
You can already use `x is >= 1 and <= 10` today.

**Reply by giorgi02** [reply]
yes I know that, but this `1 <= x <= 10` would be more intuitive, descriptive and short

**Reply by jnm2** [reply]
The other limitation is that patterns can only compare against constants.

**Comment by HaloFour** [comment, marked as answer]
See: #4108
